app/controllers/posts_controller.rb

```diff
@@ -50,6 +50,21 @@ def interactions
     end
   end
 
+  def mentionable_in_comment
+    respond_to do |format|
+      format.json {
+        if params[:id].present? && params[:q].present?
```

Comment:
I would prefer a solution with the available data from the frontend (because everything should already be there), but if that isn't possible or too complicated, then a query without q parameter to the backend should be easy.